raiden/channel/participant_state.py

```diff
@@ -3,44 +3,40 @@
 
 from raiden.mtree import Merkletree
 from raiden.utils import sha3
-from raiden.channel.balance_proof import BalanceProof
 
 log = slogging.getLogger(__name__)  # pylint: disable=invalid-name
 
 
 class ChannelEndState(object):
     """ Tracks the state of one of the participants in a channel. """
 
-    def __init__(
-            self,
-            participant_address,
-            participant_balance,
-            opened_block,
-            transferred_amount=0):
+    def __init__(self,
+                 participant_address,
+                 participant_balance,
+                 balance_proof):
+
         # since ethereum only uses integral values we cannot use float/Decimal
         if not isinstance(participant_balance, (int, long)):
             raise ValueError('participant_balance must be an integer.')
 
-        if not opened_block > 0:
-            raise ValueError('opened_block must be non-zero and positive (not %s)' % opened_block)
-
         self.contract_balance = participant_balance
         self.address = participant_address
 
-        # amount of token transferred and unlocked
-        self.transferred_amount = transferred_amount
-
-        # Sequential nonce, current value has not been used, 0 must not be used
-        # since in the netting contract it represents null.
-        #
-        # The nonce value must be inside the netting channel allowed range
-        # that is defined in terms of the opened block
-        self.nonce = 1 * (opened_block * (2 ** 32))
-
         # contains the last known message with a valid signature and
         # transferred_amount, the secrets revealed since that transfer, and the
         # pending locks
-        self.balance_proof = BalanceProof()
+        self.balance_proof = balance_proof
+
+    def transferred_amount(self, other):
```

Comment:
Will the `other` participant in a participant state ever change? Since it won't why should we need to provide it every time here? Perhaps rework the interface so that this argument does not need to be provided?

Reply:
Disregard the above comment. Saw that Konrad made the same comment and that you address it in: https://github.com/raiden-network/raiden/pull/932/files